pyhdfe/algorithms.py

```diff
@@ -188,21 +201,22 @@ def __init__(
         super().__init__(ids, cluster_ids, drop_singletons, compute_degrees, degrees_method)
         self._D = np.hstack([g.dense_dummies(drop_last=i > 0) for i, g in enumerate(self._groups_list)])
 
-    def _residualize_matrix(self, matrix: Array) -> Array:
+    def _residualize_matrix(self, matrix: Array, weights: Union[Array, None]) -> Array:
         """Compute residuals from regressions of each matrix column on the dummy variables."""
+        if weights is not None:
```

Comment:
Great, yeah, something like this for each non-supported algorithm. Adding a test that this works too, would be useful. I.e. we should get an exception for each non-supported algorithm / option config.

Reply:
Fixed all cases where weights are not supported to draw a `NotImplementedError`.